src/pipelinedefinition.hpp

```diff
@@ -129,6 +129,7 @@ class PipelineDefinition {
     Status validateNodes(ModelManager& manager);
     Status validateForCycles();
     const std::string& getName() const { return pipelineName; }
+    const PipelineDefinitionStateCode getStatusCode() const { return status.getStateCode(); }
```

Comment:
By `state` and `status` you mean something different? 
Method name is called getStatusCode and returned value has type PipelineDefinition**StateCode**. 
Is it possible to rename one of this to avoid confusion?